packages/ui/lit-grid/src/playwright/dx-grid.spec.ts

```diff
@@ -0,0 +1,217 @@
+  browser: Browser;
+
+  async open() {
+    const { page } = await setupPage(this.browser, {
+      url: storybookUrl('dx-grid--spec'),
+      viewportSize: {
+        width: (gridPlaneCellSize + gap) * (nCols + 1.5),
+        height: (gridPlaneCellSize + gap) * (nRows + 1.5),
+      }, // 336 x 272
+    });
+    this.page = page;
+    return this.page.locator('.dx-grid').waitFor({ state: 'visible' });
+  }
+
+  close() {
+    return this.page?.close();
+  }
+
+  planes() {
+    return this.page!.locator('.dx-grid [data-dx-grid-plane]').all();
+  }
+
+  cellsWithinPlane(plane: string) {
+    return this.page!.locator(`.dx-grid [data-dx-grid-plane="${plane}"]`).getByRole('gridcell').all();
+  }
+
+  cell(col: number, row: number, plane: string) {
+    return this.page!.locator(
+      `.dx-grid [data-dx-grid-plane="${plane}"] [aria-colindex="${col}"][aria-rowindex="${row}"]`,
+    );
+  }
+
+  panByWheel(deltaX: number, deltaY: number) {
+    return this.page!.locator('.dx-grid [data-dx-grid-plane="grid"]').dispatchEvent('wheel', { deltaX, deltaY });
+  }
+
+  async forCellsInRange(
+    start: DxGridPlanePosition,
+    end: DxGridPlanePosition,
+    iterator: (col: number, row: number) => Promise<void>,
+  ) {
+    const nCols = 1 + end.col - start.col;
+    const nRows = 1 + end.row - start.row;
+
+    await Promise.all(
+      [...Array(nCols)].map(async (_, c0) => {
+        return Promise.all(
+          [...Array(nRows)].map(async (_, r0) => {
+            return iterator(start.col + c0, start.row + r0);
+          }),
+        );
+      }),
+    );
+  }
+
+  async expectSelectionResult(start: DxGridPlanePosition, end: DxGridPlanePosition) {
+    return this.forCellsInRange(start, end, (col, row) =>
+      expect(this.cell(col, row, 'grid')).toHaveAttribute('aria-selected', 'true'),
+    );
+  }
+
+  async expectVirtualizationResult(cols: number, rows: number, minColIndex = 0, minRowIndex = 0) {
+    await this.cell(minColIndex, minRowIndex, 'grid').waitFor({ state: 'visible' });
+    // Top planes
+    await expect(await this.cellsWithinPlane('fixedStartStart')).toHaveLength(4);
+    await expect(await this.cellsWithinPlane('frozenRowsStart')).toHaveLength(2 * cols);
+    await expect(await this.cellsWithinPlane('fixedStartEnd')).toHaveLength(2);
+    // Center planes
+    await expect(await this.cellsWithinPlane('frozenColsStart')).toHaveLength(2 * rows);
+    await expect(await this.cellsWithinPlane('grid')).toHaveLength(rows * cols);
+    await expect(await this.cellsWithinPlane('frozenColsEnd')).toHaveLength(rows);
+    // Bottom planes
+    await expect(await this.cellsWithinPlane('fixedEndStart')).toHaveLength(2);
+    await expect(await this.cellsWithinPlane('frozenRowsEnd')).toHaveLength(cols);
+    await expect(await this.cellsWithinPlane('fixedEndEnd')).toHaveLength(1);
+  }
+
+  async expectFocus(locator: Locator) {
+    await expect(await locator.evaluate((node) => document.activeElement === node)).toBeTruthy();
```

Comment:
```suggestion
   expect(await locator.evaluate((node) => document.activeElement === node)).toBeTruthy();
```
might be wrong but iirc this only needed to be awaited if the thing passed in was a locator and not a firm value